tommath_private.h

```diff
@@ -140,22 +140,29 @@ typedef uint64_t mp_word;
 
 MP_STATIC_ASSERT(correct_word_size, sizeof(mp_word) == (2u * sizeof(mp_digit)))
 
-/* default precision */
-#ifndef MP_PREC
+/* default number of digits */
+#ifndef MP_DEFAULT_SIZE
 #   ifndef MP_LOW_MEM
-#      define MP_PREC 32        /* default digits of precision */
+#      define MP_DEFAULT_SIZE 32
 #   else
-#      define MP_PREC 8         /* default digits of precision */
+#      define MP_DEFAULT_SIZE 8
 #   endif
 #endif
 
-/* Minimum number of available digits in mp_int, MP_PREC >= MP_MIN_PREC
+/* Minimum number of available digits in mp_int, MP_DEFAULT_SIZE >= MP_MIN_SIZE
  * - Must be at least 3 for s_mp_div_school.
  * - Must be large enough such that the mp_set_u64 setter can
  *   store uint64_t in the mp_int without growing
  */
-#define MP_MIN_PREC MP_MAX(3, (((int)MP_SIZEOF_BITS(uint64_t) + MP_DIGIT_BIT) - 1) / MP_DIGIT_BIT)
-MP_STATIC_ASSERT(prec_geq_min_prec, MP_PREC >= MP_MIN_PREC)
+#define MP_MIN_SIZE MP_MAX(3, (((int)MP_SIZEOF_BITS(uint64_t) + MP_DIGIT_BIT) - 1) / MP_DIGIT_BIT)
+MP_STATIC_ASSERT(prec_geq_min_prec, MP_DEFAULT_SIZE >= MP_MIN_SIZE)
+
+/* Maximum number of digits.
+ * - Must be small enough such that mp_bit_count does not overflow.
+ * - Must be small enough such that mp_radix_size for base 2 does not overflow.
+ *   mp_radix_size needs two additional bytes for zero termination and sign.
+ */
+#define MP_MAX_SIZE ((INT_MAX - 2) / MP_DIGIT_BIT)
```

Comment:
`MP_MAX_SIZE` is, although only internally used, a bit nondescript. It is the maximum number of limbs allowed, so something in the line of `MP_MAX_LIMB_COUNT` maybe?

Reply:
We could also go back to MAX_PREC if that's desired.

Reply:
MAX_DIGIT_COUNT, DEFAULT_DIGIT_COUNT, MIN_DIGIT_COUNT?

Reply:
I like `MAX_DIGIT_COUNT` (`MP_MAX_DIGIT_COUNT`?).

You don't have to change it, but I found `MP_MAX_SIZE` a bit too ambiguous.